apps/actions-dependencies-updater/src/updater.mts

```diff
@@ -0,0 +1,192 @@
+import * as log from "./logger.mjs";
+import { RunContext } from "./index.mjs";
+import {
+  Action,
+  WorkflowByName,
+  extractDetailsFromActionIdentifier,
+} from "./workflows.mjs";
+
+import * as github from "./github.mjs";
+import * as git from "./git-cli.mjs";
+
+import { readFile, writeFile } from "node:fs/promises";
+import { join } from "node:path";
+import { getActionYamlPath } from "./utils.mjs";
+
+export interface UpdateTransaction {
+  [key: string]: {
+    references: {
+      file: string;
+      ref: string;
+    }[];
+    newVersion: {
+      sha: string;
+      version: string;
+    };
+  };
+}
+
+export async function compileUpdates(
+  ctx: RunContext,
+  workflowsByName: WorkflowByName,
+) {
+  // iterate over workflows
+  for (const workflow of Object.values(workflowsByName)) {
+    log.debug(`Workflow: ${workflow.name} - path: ${workflow.path}`);
+
+    for (const job of workflow.jobs) {
+      log.debug(`Job: ${job.name}`);
+
+      for (const dependency of job.directDependencies) {
+        await processActionDependency(ctx, dependency, workflow.path);
```

Comment:
Do we have to worry about this being serially executed? If so, we could process this concurrently instead via `Promise.all`

Reply:
Added concurrency.